resources/lib/putio.py

```diff
@@ -205,15 +205,30 @@ def subtitles(self):
             xbmcvfs.mkdirs(subtitle_directory_path)
 
             subtitle_name = subtitle['name']
-            subtitle_language = subtitle['language_code'] or ''
-            if len(subtitle_language) == 3:  # sometimes this returns wrong
-                subtitle_name = subtitle_name + ' - ' + subtitle_language
+            if subtitle_name.endswith('.srt'):
+                subtitle_name = subtitle_name[:-4]
+
+            subtitle_lang = subtitle['language_code'] or ''
+            if len(subtitle_lang) not in (2, 3):  # sometimes this returns wrong
+                subtitle_lang = 'und'
+
+            # language is unknown for folder subtitles, use last part if name ends with `.` + 2-3 chars
+            if subtitle['source'] == 'folder':
```

Comment:
Note that even if a subtitle with source `folder` has a valid `language_code` which is *not* `und`, it will be (possibly) overwritten with a locally parsed "language code" here. While the putio api may only return `und` for `folder`-sourced subtitles *today*, this is not a future-proof assumption.

An improvement would be to only perform filename parsing if the `language_code` is empty or `und`.